File: GourmetiseMobile/app/src/main/java/com/example/appgourmetiseconcours/BakeryDAO.java
package com.example.appgourmetiseconcours;

import android.content.ContentValues;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

public class BakeryDAO {
    private SQLiteDatabase db;

    public BakeryDAO(Context context) {
        BakeryHelper helper = new BakeryHelper(context);
        db = helper.getWritableDatabase();
    }

    public void insertBakery(String name, String street, String postcode, String city, String phonenumber, String contactname, String description) {
        ContentValues values = new ContentValues();
        values.put("name", name);
        values.put("street", street);
        values.put("postcode", postcode);
        values.put("city", city);
        values.put("phonenumber", phonenumber);
        values.put("contactname", contactname);
        values.put("description", description);
        db.insert("bakery", null, values);
    }

    // Define methods for updating, deleting, and querying bakeries as needed
}

File: GourmetiseMobile/app/src/main/java/com/example/appgourmetiseconcours/BakeryDAO.kt
package com.example.appgourmetiseconcours

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase

class BakeryDAO(context: Context) {
    private val db: SQLiteDatabase = BakeryHelper(context).writableDatabase

    fun insertBakery(siren: String, name: String, street: String, postcode: String, city: String, phonenumber: String, contactname: String, description: String?) {
        val values = ContentValues().apply {
            put("siren", siren)
            put("name", name)
            put("street", street)
            put("postcode", postcode)
            put("city", city)
            put("phonenumber", phonenumber)
            put("contactname", contactname)
            put("description", description)
        }
        db.insert("bakery", null, values)
    }

    fun clearAllBakeries() {
        db.delete("bakery", null, null)
    }

    // Define methods for updating, deleting, and querying bakeries as needed
}
